tests/vm_generator.go

```diff
@@ -0,0 +1,95 @@
+package tests
+
+import (
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
+func randName(name string) string {
+	return name + "-" + rand.String(5)
+}
+
+func getBaseVMISpec() *v1.VirtualMachineInstanceSpec {
+	return &v1.VirtualMachineInstanceSpec{
+		Domain: v1.DomainSpec{
+			Resources: v1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceMemory: resource.MustParse("64M"),
+				},
+			},
+		},
+	}
+}
+
+func addContainerDisk(spec *v1.VirtualMachineInstanceSpec, image string) *v1.VirtualMachineInstanceSpec {
+	disk := &v1.Disk{
+		Name: "containerdisk",
+		DiskDevice: v1.DiskDevice{
+			Disk: &v1.DiskTarget{
+				Bus: "virtio",
+			},
+		},
+	}
+	spec.Domain.Devices.Disks = append(spec.Domain.Devices.Disks, *disk)
+	volume := &v1.Volume{
+		Name: "containerdisk",
+		VolumeSource: v1.VolumeSource{
+			ContainerDisk: &v1.ContainerDiskSource{
+				Image: image,
+			},
+		},
+	}
+	spec.Volumes = append(spec.Volumes, *volume)
+	return spec
+}
+
+func getBaseVMI(name string) *v1.VirtualMachineInstance {
+	baseVMISpec := getBaseVMISpec()
+
+	return &v1.VirtualMachineInstance{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1.GroupVersion.String(),
+			Kind:       "VirtualMachineInstance",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"special": name},
```

Comment:
why do we need this lablel?

Reply:
All kubevirt example vmis have this label. I"m not sure why/if it is indeed needed.

Reply:
lets remove it then please? we don't need it